common/frame_session.go

```diff
@@ -749,6 +749,33 @@ func (fs *FrameSession) onFrameNavigated(frame *cdp.Frame, initial bool) {
 			fs.ctx, fs.targetID.String(), frame.URL,
 			trace.WithAttributes(attribute.String("url", frame.URL)),
 		)
+
+		var (
+			spanID   = fs.mainFrameSpan.SpanContext().SpanID().String()
+			newFrame = fs.manager.getFrameByID(frame.ID)
+		)
+		// Set k6SpanId property in the page so it can be retrieved when pushing
+		// the Web Vitals events from the page execution context and used to
+		// correlate them with the navigation span to which they belong to.
+		if newFrame != nil {
+			// Executing a CDP command in the event parsing goroutine might deadlock in some cases.
+			// For example a deadlock happens if the content loaded in the frame that has navigated
+			// includes a JavaScript initiated dialog which we have to explicitly accept or dismiss
+			// (see onEventJavascriptDialogOpening). In that case our EvaluateGlobal call can't be
+			// executed, as the browser is waiting for us to accept/dismiss the JS dialog, but we
+			// can't act on that because the event parsing goroutine is stuck in onFrameNavigated.
+			// Because in this case the action is to set an attribute to the global object (window)
+			// it should be safe to just execute this in a separate goroutine.
+			go func() {
+				js := fmt.Sprintf("window.k6SpanId = '%s';", spanID)
```

Comment:
Initially I was thinking that considering that this is set in the context of the web page, I don't think it's gonna collide with other tracing related functionalities that could be implemented in k6, and also that it's an internal implementation to k6 browser, we can change it at any time, so I would prefer to keep it simple and shorter as of now. But I'm thinking this might be referenced by the Faro integration, so... maybe we could add the `browser` naming there, eve though I don't see having an collisions in the page context. What are your thoughts also @inancgumus ?

Reply:
Rethinking this, this is a completely internal mechanism for k6 browser in order to correlate traces, so no integration should take this into account as a reference, instead, the _test run id_ should be used. So I think we can keep this simple and short by now.